### DISCUSSION TITLE
Enriching extensions for collections

### DISCUSSION THREAD

**Discussion by khalidabuhakmeh** [11ty/eleventy#4128] [Enhancement Queue]
### Is your feature request related to a problem? Please describe.

I would like to "enrich" items as they get added to collections without creating a brand new collection using the `addCollection` method.
 

### Describe the solution you'd like

I want to write an enriching function similar to the examples below. These are my initial thoughts, and I'm unsure if items would be tracked across each collection or if each has a unique copy of an item. I'm also not sure if users should have access to more than the current item and the collection metadata it will be added to.

```javascript
elventyConfig.onCollectionItemAdding(async (collection, item) => {
    const id = item.id;
    if (item.databaseType == 'Car') {
        const available
            = await db.query(`select * from cars where brand = '${item.brand}`);
        item.available = mapToAvailableCars(available); 
    }
    // return the item to add it
    item;
});
```

```javascript
eleventConfig.onCollectionItemAdding(async (collection, item) => {
    const available = db.showItemOnSite(item.id)
    item.enabled = available;
    return item;
});
```

```javascript
eleventConfig.onCollectionItemAdding(async (collection, item) => {
    if (collection.name == 'UsStates')
    {
        const response = await EleventyFetch('https://example.com/demographics/${state.id}', {
            type: 'json',
            duration: '1d'
        });
        item.population = demographics.population;
    }
    return item;
});
```

**Comment by zachleat** [comment]
Can this problem be solved with Computed Data? https://www.11ty.dev/docs/data-computed/
